src/components/ReportWelcomeText.tsx

```diff
@@ -1,114 +1,86 @@
+        // @ts-expect-error TODO: Remove this once OptionsListUtils (https://github.com/Expensify/App/issues/24921) is migrated to TypeScript.
+        OptionsListUtils.getPersonalDetailsForAccountIDs(participantAccountIDs, personalDetails),
         isMultipleParticipant,
     );
-    const isUserPolicyAdmin = PolicyUtils.isPolicyAdmin(props.policy);
-    const roomWelcomeMessage = ReportUtils.getRoomWelcomeMessage(props.report, isUserPolicyAdmin);
-    const moneyRequestOptions = ReportUtils.getMoneyRequestOptions(props.report, participantAccountIDs);
+    const isUserPolicyAdmin = PolicyUtils.isPolicyAdmin(policy);
+    const roomWelcomeMessage = ReportUtils.getRoomWelcomeMessage(report, isUserPolicyAdmin);
+    const moneyRequestOptions = ReportUtils.getMoneyRequestOptions(report, participantAccountIDs);
 
     return (
         <>
             <View>
                 <Text style={[styles.textHero]}>
-                    {isChatRoom ? props.translate('reportActionsView.welcomeToRoom', {roomName: ReportUtils.getReportName(props.report)}) : props.translate('reportActionsView.sayHello')}
+                    {isChatRoom ? translate('reportActionsView.welcomeToRoom', {roomName: ReportUtils.getReportName(report)}) : translate('reportActionsView.sayHello')}
                 </Text>
             </View>
             <Text style={[styles.mt3, styles.mw100]}>
                 {isPolicyExpenseChat && (
                     <>
-                        <Text>{props.translate('reportActionsView.beginningOfChatHistoryPolicyExpenseChatPartOne')}</Text>
-                        <Text style={[styles.textStrong]}>{ReportUtils.getDisplayNameForParticipant(props.report.ownerAccountID)}</Text>
-                        <Text>{props.translate('reportActionsView.beginningOfChatHistoryPolicyExpenseChatPartTwo')}</Text>
-                        <Text style={[styles.textStrong]}>{ReportUtils.getPolicyName(props.report)}</Text>
-                        <Text>{props.translate('reportActionsView.beginningOfChatHistoryPolicyExpenseChatPartThree')}</Text>
+                        <Text>{translate('reportActionsView.beginningOfChatHistoryPolicyExpenseChatPartOne')}</Text>
+                        <Text style={[styles.textStrong]}>{ReportUtils.getDisplayNameForParticipant(report?.ownerAccountID)}</Text>
+                        <Text>{translate('reportActionsView.beginningOfChatHistoryPolicyExpenseChatPartTwo')}</Text>
+                        <Text style={[styles.textStrong]}>{ReportUtils.getPolicyName(report)}</Text>
+                        <Text>{translate('reportActionsView.beginningOfChatHistoryPolicyExpenseChatPartThree')}</Text>
                     </>
                 )}
                 {isChatRoom && (
                     <>
                         <Text>{roomWelcomeMessage.phrase1}</Text>
-                        {roomWelcomeMessage.showReportName && (
+                        {roomWelcomeMessage.showReportName && !!report?.reportID && (
```

Comment:
`ROUTES.REPORT_WITH_ID_DETAILS.getRoute` requires string, not undefined type, so `!!report?.reportID` is added here for the conditional rendering

Reply:
@nikosamofa I think is better to do this condition inside `onPress` callback instead of here, e.g.:

```ts
onPress={() => {
    if (!report?.reportID) {
        return;
    }

    Navigation.navigate(ROUTES.REPORT_WITH_ID_DETAILS.getRoute(report.reportID));
}}
```

You can also extract this `onPress` callback to a separate function for better code visibility.

Reply:
Addressed by [73b8b7f](https://github.com/Expensify/App/pull/33251/commits/73b8b7f3345fda6344a09e58e6fa450eb9346ba0)